lib/benchee/profile.ex

```diff
@@ -0,0 +1,63 @@
+defmodule Benchee.Profile do
+  @moduledoc """
+  Profiles each scenario after benchmarking them if the `profile_after` flag is set to `true`.
+
+  The profiler that will be used is either the one set by the `profiler` option
+  in the `profile` configuration or the default one (`:cprof`). It accepts however the following
+  profiler options:
+    * `:cprof` will profile with `Mix.Task.Profile.Cprof` (useful to discover bottlenecks
+    related to function calls).
+    * `:eprof` will profile with `Mix.Task.Profile.Eprof` (useful to discover bottlenecks
+    related to time information of each function call).
+    * `:fprof` will profile with `Mix.Task.Profile.Fprof` (useful to discover bottlenecks
+    of a sequential code).
+  """
+  alias Benchee.Suite
+
+  @doc """
+  Runs for each scenario found in the suite the `profile/2` function from the given profiler.
+  """
+  @spec profile(Suite.t()) :: Suite.t()
+  def profile(suite = %{configuration: %{profile: %{profile_after: false}}}), do: suite
+
+  def profile(
+        suite = %{
+          configuration: %{profile: %{profiler: profiler, profiler_opts: profiler_opts}},
+          scenarios: scenarios
+        }
+      ) do
+    profiler_module = profiler_to_module(profiler)
+
+    Enum.each(scenarios, &run(&1, {profiler, profiler_module, profiler_opts}))
+
+    suite
+  end
+
+  defp run(%{name: name, function: fun_to_profile}, {profiler, profiler_module, profiler_opts}) do
+    IO.puts("\nProfiling #{name} with #{profiler}...")
```

Comment:
This would always show up in our test output, the way we deal with this usually is to have a separate printer to delegate this work to - have a look at benchmark printer for instance: https://github.com/bencheeorg/benchee/blob/master/lib/benchee/output/benchmark_printer.ex

Reply:
After looking at the benchmark printer I can understand delegating the printing to it, as there is a lot of prints to do regarding benchmarking (and options to disable each of them) and otherwise it would clutter the rest of modules, but do we really care about this one line, which would _only_ get printed (as a separator between the profiling of each scenario) if profiling is enabled, to show up on the test output? The existing regex won't pick it up and make some test fail, if that's the concern.

Reply:
The regex testing isn't the concern. The concern is that one line showing up during test runs (repeatedly) once unit tests are written. The overall tests capture io mostly anyhow.

I like clean test run output. Having just one/a couple of tests allowed to print out creates a broken window syndrome kinda situation where now practically every test would be allowed to do this and it'd happen faster than you'd think :) 

So yeah, a separate printing module would be much preferred.